PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,542 @@
+                self.cleanUp(callEnviron)
+                return pos
+
+            # if the value hasn't been found, wait and then move the arrow over one cell
+            self.wait(0.7)
+            cur = cur.next
+           
+            arrow = (arrow, )
+            if (pos-4)%self.LEN_ROW == 0: 
+                self.moveItemsBy(arrow, (-((self.CELL_WIDTH + self.CELL_GAP)*(self.LEN_ROW-1)), (self.CELL_HEIGHT + self.ROW_GAP)))
+            else: 
+                self.moveItemsBy(arrow, (self.CELL_WIDTH + self.CELL_GAP,0))
+            pos+=1
+            
+        self.cleanUp(callEnviron)
+        return None
+            
+    
+    ### BUTTON FUNCTIONS##
+    def clickSearch(self):
+        val = self.validateArgument()
+        if val is None:
+            self.setMessage("Input value must be less than 9 characters")
+        else:
+            result = self.search(val)
+            if result != None:
+                msg = "Found {}!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+            self.setMessage(msg)
+            
+        self.clearArgument()
+    
+    def clickInsert(self):
+        val = self.validateArgument()
+        if val is None:
+            self.setMessage("Input value must be less than 9 characters")
+        elif len(self) >= self.MAX_SIZE:
+            self.setMessage("Error! Linked List is already full.")
+        else:  
+            self.insertElem(val)
+
+    def clickDelete(self):
+        val = self.validateArgument()
+        if not self.first:
+            msg = "ERROR: Linked list is empty"
+        elif not val:
+            msg = "Input value must be less than 9 characters" 
+        else:
+            result = self.delete(val)
+            if result != None:
+                msg = "{} Deleted!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+        self.setMessage(msg)
+        self.clearArgument()
+        
+    def clickDeleteFirst(self):
+        val = self.validateArgument()
+        result = self.deleteFirst()
+        if not self.first: 
+            msg = "ERROR: Linked list is empty"
+        else:
+            msg = "first node deleted"
+        self.setMessage(msg)
+        self.clearArgument()
+        
+        
+    def clickNewLinkedList(self):
+        self.canvas.delete('all')
+        self.newLinkedList()
+    
+    def clickGetFirst(self):
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
+    
+
+    def makeButtons(self):
+        vcmd = (self.window.register(validate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        searchButton = self.addOperation(
+            "Search", lambda: self.clickFind(), numArguments=1,
+            validationCmd=vcmd)
```

Comment:
I think this is supposed to call `clickSearch()`, not `clickFind()`.

By the way, you can simplify the `lambda: self.clickSearch()` expression to just `self.clickSearch`.  That will call the `clickSearch()` method to be called with no arguments (its `self` argument is automatically handled by Python).